pallets/allocations/src/lib.rs

```diff
@@ -109,13 +112,9 @@ pub mod pallet {
 				return Ok(Pays::No.into());
 			}
 
-			let coins_already_allocated = Self::coins_consumed();
-			let coins_that_will_be_consumed = coins_already_allocated
-				.checked_add(&amount)
-				.ok_or("Overflow computing coins consumed")?;
-
+			let current_supply = T::Currency::total_issuance();
 			ensure!(
-				coins_that_will_be_consumed <= T::MaximumCoinsEverAllocated::get(),
+				current_supply.saturating_add(amount) <= T::MaximumSupply::get(),
 				Error::<T>::TooManyCoinsToAllocate
 			);
```

Comment:
I would love a per block, or per time window limit. Though it would be more of a new feature VS an optimization.

The original thinking (re-issuance) was to provide a guarantee we'd never go over 21B. I had originally computed the "magic value" `MaximumCoinsEverAllocated` myself to guarantee this, the current changes simply make it more expressive, and easier to verify